composer/trainer/trainer.py

```diff
@@ -526,15 +601,54 @@ def __init__(
                                                        optimizers=self.state.optimizers)
 
             # Move any remaining optimizer parameters onto the device
-            self.state.optimizers = map_collection(self.state.optimizers, self.device.optimizer_to_device)
+            self.state.optimizers = map_collection(self.state.optimizers, self._device.optimizer_to_device)
 
             if dist.get_world_size() > 1:
                 # Only wrap the module if required
-                self.state.model = prepare_ddp_module(self.state.model, self.find_unused_parameters)
+                self.state.model = prepare_ddp_module(self.state.model, self._find_unused_parameters)
 
     @property
-    def deepspeed_enabled(self):
-        return self.deepspeed_config is not None
+    def deepspeed_enabled(self) -> bool:
+        """``True`` if DeepSpeed is being used for training and ``False`` otherwise.
+
+        .. seealso:: `DeepSpeed's documentation <https://www.deepspeed.ai/docs/config-json/>`_
+        """
+        return self._deepspeed_config is not None
+
+    @property
+    def saved_checkpoints(self) -> Dict[Timestamp, List[str]]:
+        """The times and paths to checkpoint files saved across all ranks during training.
+
+        .. note:: When using DeepSpeed, the index of a filepath in each list corresponds to the
```

Comment:
maybe move the note into the return docstring? It's a bit confusing without knowning the return format.

```suggestion
        .. note:: When using DeepSpeed, the index of a filepath corresponds to the
```